### DISCUSSION TITLE
Support for elastic Plastic FE- Calculation

### DISCUSSION THREAD

**Discussion by ThumserTKP** [boschresearch/pylife#46] [Ideas]
The fkm-nonlinear guideline evaluates the elastic plastic behaviour based on an elastic solution and notch approximation relationship like Neuber or Beste / Seeger. For some cases there the material response should directly be caluclated by aid for finete element methode, e.g. with high nonlienearities as contact. For this cas it would be nice if pylife can import and us it directly from finite element caluclation.

**Comment by johannes-mueller** [comment]
@maierbn maybe you want to comment.

**Comment by maierbn** [comment]
Yes indeed. Pylife has functionality to load Abaqus and Ansys FEM results. But replacing the notch approximation in the FKM nl algorithm is not supported. We're thinking about extending it in this direction, but it will take some time.

**Comment by KristinaLepper** [comment]
@ThumserTKP , I like the idea of using elastic-plastic FEM and making the notch approximation obsolete. Maybe this could also give new possibilities to deal with non-proportional loading. 
From coding point of view, I guess it's really doable to jump right to the point where we have the turning points of the stress-strain hysereses and then calculate the damage values (P_RAM or P_RAJ). But how do we get from the stress and strain tensors of the FEM to a decent stress-strain path? A signed von Mises stress (and strain) will probably give a chaotic zig-zag instead of a hysteresis.

**Comment by ThumserTKP** [comment]
@KristinaLepper, For the current state of the guideline only uniaxial or multiaxial propotional loading is valid, see table 2.1 of the guideline. So for this special case the use of signed von Mises values will be result in a none chaotic behaviour. 
I think the non- propotional loading is an huge amount of work to deal correclty with it. For non-propotional loading I suggest to review the PHD thesises done at Darmstadt: https://www.werkstoffmechanik.tu-darmstadt.de/forschung_wm/publikationen_wm/institutsveroeffentlichungen_wm/index.de.jsp.
